compiler/rustc_typeck/src/astconv/generics.rs

```diff
@@ -57,19 +59,43 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
             GenericArg::Const(_) => ParamKindOrd::Const { unordered },
         };
 
-        if matches!(arg, GenericArg::Type(hir::Ty { kind: hir::TyKind::Path { .. }, .. }))
-            && matches!(kind_ord, ParamKindOrd::Const { .. })
-        {
-            let suggestions = vec![
-                (arg.span().shrink_to_lo(), String::from("{ ")),
-                (arg.span().shrink_to_hi(), String::from(" }")),
-            ];
-            err.multipart_suggestion(
-                "if this generic argument was intended as a const parameter, \
-                try surrounding it with braces:",
-                suggestions,
-                Applicability::MaybeIncorrect,
-            );
+        // Specific suggestion set for diagnostics
+        if let Some(param) = param {
+            match (arg, &param.kind) {
+                (
+                    GenericArg::Type(hir::Ty { kind: hir::TyKind::Path { .. }, .. }),
+                    GenericParamDefKind::Const { .. },
+                ) => {
+                    let suggestions = vec![
+                        (arg.span().shrink_to_lo(), String::from("{ ")),
+                        (arg.span().shrink_to_hi(), String::from(" }")),
+                    ];
+                    err.multipart_suggestion(
+                        "if this generic argument was intended as a const parameter, \
+                    try surrounding it with braces:",
+                        suggestions,
+                        Applicability::MaybeIncorrect,
+                    );
+                }
+                (
+                    GenericArg::Type(hir::Ty { kind: hir::TyKind::Array(_, len), .. }),
+                    GenericParamDefKind::Const { .. },
+                ) => {
+                    if tcx.type_of(param.def_id).is_ptr_sized_integral() {
```

Comment:
```suggestion
                    if tcx.type_of(param.def_id) == tcx.types.usize {
```
not sure if emitting this help for `fn test<const N: isize>()` is helpful.

Reply:
ah I didn't know that existed so I just picked the function that looked the closest, definitely `isize` is not necessary to report